processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -1154,22 +1220,23 @@ protected ColumnSelectorFactory makeColumnSelectorFactory(
 
   protected final Comparator<IncrementalIndexRow> dimsComparator()
   {
-    return new IncrementalIndexRowComparator(dimensionDescsList);
```

Comment:
It would be nice to mention in the PR description that you fixed a concurrency bug: there was unsynchronized access to a concurrently updated ArrayList.

Reply:
Updated :)

Reply:
For context, here is an explanation of this kind bug, including specifically in application to ArrayList: https://github.com/code-review-checklists/java-concurrency#unsafe-concurrent-point-read